crates/sozo/ops/src/auth.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::{Context, Result};
+use dojo_world::contracts::world::WorldContract;
+use dojo_world::metadata::Environment;
+use dojo_world::utils::TransactionWaiter;
+use starknet::accounts::Account;
+
+use super::get_contract_address;
+use sozo::commands::auth::ResourceType;
+
+pub async fn execute(
+    kind: String,
+    world: WorldContract,
+    starknet: StarkNet,
+    account: Account,
+    transaction: Transaction,
+    env_metadata: Option<Environment>,
+    models_contracts: Option<Vec<ModelContract>>,
+    owners_resources: Option<Vec<OwnerResource>>,
+) -> Result<()> {
+    match kind.as_str() {
+        "writer" => {
+            grant_writer(world, starknet, account, models_contracts, transaction, env_metadata).await?;
+        }
+        "owner" => {
+            grant_owner(world, starknet, account, owners_resources, transaction, env_metadata).await?;
+        }
+        _ => return Err(anyhow::anyhow!("Invalid kind, must be either 'writer' or 'owner'")),
+    }
+
+    Ok(())
+}
+
+
+pub async fn grant_writer(
+    world: WorldContract,
+    starknet: StarkNet,
+    account: Account,
+    models_contracts: Vec<ModelContract>,
+    transaction: Transaction,
+    env_metadata: Option<Environment>,
+) -> Result<()> {
+    let world_address = world.address(env_metadata.as_ref())?;
+    let provider = starknet.provider(env_metadata.as_ref())?;
+
+    let account = account.account(&provider, env_metadata.as_ref()).await?;
+    let world = WorldContract::new(world_address, &account);
+
+    let mut calls = Vec::new();
+
+    for mc in models_contracts {
+        let contract = get_contract_address(&world, mc.contract).await?;
+        calls.push(world.grant_writer_getcall(&mc.model, &contract.into()));
+    }
+
+    let res = account
+        .execute(calls)
+        .send()
+        .await
+        .with_context(|| "Failed to send transaction")?;
+
+    if transaction.wait {
+        let receipt = TransactionWaiter::new(res.transaction_hash, &provider).await?;
+        println!("{}", serde_json::to_string_pretty(&receipt)?);
+    } else {
+        println!("Transaction hash: {:#x}", res.transaction_hash);
+    }
+
+    Ok(())
+}
+
+pub async fn grant_owner(
+    world: WorldContract,
+    starknet: StarkNet,
+    account: Account,
+    owners_resources: Vec<OwnerResource>,
+    transaction: Transaction,
+    env_metadata: Option<Environment>,
+) -> Result<()> {
+    let world_address = world.address(env_metadata.as_ref())?;
+    let provider = starknet.provider(env_metadata.as_ref())?;
+
+    let account = account.account(&provider, env_metadata.as_ref()).await?;
+    let world = WorldContract::new(world_address, &account);
+
+    let mut calls = Vec::new();
+
+    for or in owners_resources {
+        let resource = match &or.resource {
+            ResourceType::Model(name) => *name,
```

Comment:
@glihm This is the dependency on ResourceType enum that I don't know how to resolve.

Reply:
In this case, as the `ResourceType` is a simple type (is not directly bound to `clap` attributes), you can have it declared into `ops`. And `commands` will import it to adjust the parsing of inputs.